calcom/cal.com
# ci: Add new release actions

## What does this PR do?

This pull request introduces new GitHub Actions workflows for handling draft releases and post-release tasks, and removes an obsolete release workflow. The changes aim to improve the release process by automating version bumps, commits, and pushing changes to production.

New workflows:

* [`.github/workflows/draft-release.yml`](diffhunk://#diff-3d80a0bd2aa867253847d98ff0fddb8982d61587bee3ddfb9906f4aafbe55329R1-R50): Added a new workflow for creating draft releases, which includes steps for checking out code, configuring git, bumping the version, committing changes, and drafting the release.
* [`.github/workflows/post-release.yml`](diffhunk://#diff-dfb63fa587b638fcf581007b525fdfb26835766b2a2be8840d19c6422f7d0ae6R1-R25): Added a new workflow for post-release tasks, which includes steps for checking out code, configuring git, and pushing changes to the production branch.

Removed workflows:

* [`.github/workflows/release.yml`](diffhunk://#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L1-L17): Removed the old release workflow that handled push events to tags and pushed changes to the production branch.

## Mandatory Tasks


## How should this be tested?

1. Trigger the draft-release workflow manually with a version number
2. Verify that:
   - The version is bumped correctly in apps/web
   - A commit is created with the version change
   - A draft release is created with generated release notes
3. After publishing the release, verify that:
   - The post-release workflow is triggered
   - Changes are successfully pushed to the production branch

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings
